Makefile

```diff
@@ -108,9 +108,13 @@ fmt: ## Run go fmt against code.
 vet: ## Run go vet against code.
 	go vet ./...
 
-.PHONY: test
-test: manifests generate fmt vet envtest ## Run tests.
-	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" go test -race -v ./... -coverprofile cover.out
+.PHONY: unit-test
+unit-test: manifests generate fmt vet envtest ## Run unit-test only
+	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" go test -short -race -v $$(go list ./... | grep -v /tests/e2e) -coverprofile cover.out
+
+.PHONY: e2e-test
+e2e-test: manifests generate fmt vet envtest ## Run e2e-test only
+	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" go test -race -v ./tests/e2e/... -coverprofile cover.out
```

Comment:
I don't think we need to include `-race` or `-coverprofile cover.out` here - that would be trying to create race conditions for the test itself, and code coverage for the test itself